### ISSUE TITLE
Delay JS add compatibility exclusion for PDF Embedder Plugin

### ISSUE THREAD

**Issue by markllego** [wp-media/wp-rocket#4340]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version
WP Rocket 3.9.3
- Used the search feature to ensure that the bug hasn’t been reported before
I have read the ff: https://docs.wp-rocket.me/article/1560-delay-javascript-execution-compatibility-exclusions

**Describe the bug**
When the WP-Rocket Delay JS Feature is enabled, the PDF Embedder plugin will not work as expected.

**To Reproduce**
1. Install and enable the PDF Embedder Plugin.
2. Make a post and attached a pdf file using the PDF Embedder shortcode.
3. Use Incognito mode and go to the post.
4. The embedded pdf file will fail to load, appear as a link or get stuck on loading.

**Expected behavior**
Add the following scripts to solve the issue:

/wp-content/plugins/pdf-embedder/js/all-pdfemb-basic-4.6.2.min.js,
/wp-content/plugins/pdf-embedder/js/pdfjs/pdf-4.6.2.min.js, and
/wp-content/plugins/pdf-embedder/js/pdfjs/pdf-4.6.2.worker.min.js
/wp-content/plugins/PDFEmbedder-premium/js/all-pdfemb-premium-4.4.3.min.js
/wp-content/plugins/PDFEmbedder-premium/js/pdfjs/pdf-4.4.3.min.js
/wp-content/plugins/PDFEmbedder-premium/js/pdfjs/pdf-4.4.3.worker.min.js
/wp-content/plugins/PDFEmbedder-premium-secure/js/all-pdfemb-premium-4.4.5.min.js,
/wp-content/plugins/PDFEmbedder-premium-secure/js/pdfjs/pdf-4.4.5.min.js, and
/wp-content/plugins/PDFEmbedder-premium-secure/js/pdfjs/pdf-4.4.5.worker.min.js
/wp-content/plugins/PDFEmbedder-premium-secure/js/all-pdfemb-premium-4.4.5.min.js,
/wp-content/plugins/PDFEmbedder-premium-secure/js/pdfjs/pdf-4.4.5.min.js, and
/wp-content/plugins/PDFEmbedder-premium-secure/js/pdfjs/pdf-4.4.5.worker.min.js

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by Mai-Saad** [comment]
On the test site, we will have a link to pdf till user interaction.
The following exclusions will make the PDF displayed without user interaction.
```
/pdf-embedder/js/(.*).js
jquery.min.js
jquery-migrate.min.js
```

**Comment by alfonso100** [comment]
thank you for reporting this, @markllego !

I have tested and added the exclusions for the  PDF Embedder (free version) to our [Delay JavaScript Execution compatibility exclusions](https://docs.wp-rocket.me/article/1560-delay-javascript-execution-compatibility-exclusions#pdf-embedder).

Can you please open a support request so we can check the PDFEmbedder Premium exclusions on your website? It'd be good for us to validate these, so we can make them publicly available for our users.

**Comment by Mai-Saad** [comment]
> On the test site, we will have a link to pdf

This is the case too when Combine JS is enabled (tested on WPR 3.10.2, PDF Embedder 4.6.2).

Excluding jquery and jquery-migrate will make PDF displayed 

```
jquery.min.js
jquery-migrate.min.js
```
With Defer JS, the file is stuck in loading or displayed as a link (tested on WPR 3.10.2, PDF Embedder 4.6.2).
The following exclusion fixes the problem
```
/pdf-embedder/js/(.*).js
jquery.min.js
jquery-migrate.min.js
```

**Comment by iCaspar** [comment]
PDF Embedder has been changing some of the class names that we look for to add this automatically. The most recent update of the Basic version broke our integration tests, and we have a [PR for it](https://github.com/wp-media/wp-rocket/pull/4491).

That said, if the base class for Embedder's upgraded versions has also changed, the current PR will probably not fix those versions. It looks from @markllego OP that there is an upgraded version in use there.

Going to leave this Issue open for now, until we can confirm the classnames for Premium and Secure versions.

**Comment by piotrbak** [comment]
@iCaspar I've already contacted PDF Embedder team and asked to send us copies of all premium versions they have. I believe we'll get back to this PR in the next sprint.

**Comment by iCaspar** [comment]
@piotrbak Let's merge the current PR for the basic version so we can get our test back to green for release of 3.10.3 -- We can start a new PR when we have more info on the premium versions to be in 3.10.4. (I'll unlink that PR from this issue)

**Comment by iCaspar** [comment]
Looks like we also have some CombineJS issues found by @Mai-Saad. See https://github.com/wp-media/wp-rocket/pull/4491#issuecomment-964873371

(David Bisset tells me that they are planning more revisions, so we'll have to keep an eye on this moving forward to make sure we remain compatible.)

**Comment by piotrbak** [comment]
Will be fixed in the 3.13 backend

**Comment by piotrbak** [comment]
Handled in 3.13 using dynamic lists
